pkg/controller/instance/instance_controller.go

```diff
@@ -258,6 +271,20 @@ func (r *Reconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
 	return computeTheReconcileResult(instance, time.Now), nil
 }
 
+func setReadinessOnInstance(instance *kudoapi.Instance, c client.Client) error {
```

Comment:
Would it make sense to let this func return `(boolean, error)` to check if the readiness actually changed? If it doesn't change we can skip the `updateInstance` and save an API call, or do I miss something? 

Or maybe we now that it will always change, then it doesn't make sense to add the additional logic.

Reply:
hmm yeah we can do that, maybe not this way but one level up we have instance and oldInstance 🤔 I just wonder if it's not premature optimization and if in major cases it will actually change ... but I think I'll add it as per your request